docs/datastorage/extendedentityproperties.md

```diff
@@ -0,0 +1,199 @@
+
+In order to uniquely identify your property and avoid duplication, the method takes a string parameter with an identifier for the property. A good practice is to include the modid in this string, so that it will not collide with other mods.
+
+**WARNING**: If the same property identifier is added twice, Forge will append a number to it, and return this modified identifier from the ```registerExtendedProperties``` method. If you don't want that to happen, you can use ```Entity#getExtendedProperties``` to check if an IEEP with that name was already added.
+
+Usage
+-----
+
+To make use of the extended data, the instance of the IEEP implementation has to be obtained from the Entity, and because the entity could have been unloaded or may have changed dimensions, it is not safe to cache the references.
+
+To obtain the IEEP reference, one would use Entity#getExtendedProperties, with the same property ID specified on registration. THe return value, if not null, is the instance of ```IExtendedEntityProperties``` added during entity construction.
+
+A good idea is to create a static "get" method in your IEEP implementation, that will automatically obtain the instance, and cast it to your implementation class name.
+
+Synchronizing data with clients
+-------------------------------
+
+By default, the entity data does not synchronize. In order to change this, the mods have to manage their own synchronization code using packets.
+
+There are three different situation in which you may want to send synchronization packets, all of them optional:
+
+1. When the entity spawns in the world, you may want to share the initialization-assigned values with the clients.
+1. When the stored data changes, you may want to notify some or all of the watching clients.
+1. When a new client starts viewing the entity, you may want to notify it of the existing data.
+
+Refer to the [Networking](../networking/index.md) page for more information on implementing the network packets.
+
+Persisting across player deaths
+-------------------------------
+
+By default, the entity data does not persist on death. In order to change this, the data has to be manually copied when the player entity is cloned during the respawn process.
+
+This can be done by handling the ```PlayerEvent.Clone``` event.
+
+Updating the contents over time
+-------------------------------
+
+By default, the entity data does not tick. In order to change this, the data has to be manually made to update.
+
+This can be done by handling the ```TickEvent.PlayerTickEvent``` event if the entity is attached to players.
+
+Skeleton example
+----------------
+
+This skeleton can be used as a basis for new IEEP implementations. 
+
+```Java
+public class ExampleEntityProperty implements IExtendedEntityProperties
```

Comment:
We try to avoid providing complete implementations, as this just results in copypasta. Please try to explain the concepts rather than just pasting code.